framework/src/transfers/MultiAppTransfer.C

```diff
@@ -172,6 +173,66 @@ MultiAppTransfer::checkMultiAppExecuteOn()
                        "execute_on flags"));
 }
 
+void
+MultiAppTransfer::checkPostProcessorExecuteOn(std::string to_check)
+{
+  std::vector<std::shared_ptr<MultiApp>> multi_apps;
+  if (_from_multi_app)
+    multi_apps.push_back(_from_multi_app);
+  if (_to_multi_app)
+    multi_apps.push_back(_to_multi_app);
+  for (std::shared_ptr<MultiApp> multi_app : multi_apps)
+  {
+    // check subapps
+    for (unsigned int i = 0; i < multi_app->numGlobalApps(); i++)
+    {
+      // If we do not have this app, we skip
+      if (!multi_app->hasLocalApp(i))
+        continue;
+      FEProblemBase & sub_problem = multi_app->appProblemBase(i);
+      checkAgainstPPExec(to_check, sub_problem);
+    }
+
+    // check main app
+    FEProblemBase & main_problem = multi_app->problemBase();
+    checkAgainstPPExec(to_check, main_problem);
+  }
+}
+
+void
+MultiAppTransfer::checkAgainstPPExec(std::string to_check, FEProblemBase & problem)
+{
+  // It's possible that the Transfer is going to/from a pp/vpp in a subapp. Therefore, if it doesn't
+  // have the pp/vpp name in this apps problem, return, as this behavior is checked in the subapp
+  // portion of checkPostProcessorExecuteOn.
+  if (!problem.hasUserObject(to_check))
+    return;
+
+  const ExecFlagEnum & object_execute_on = problem.getUserObjectBase(to_check).getExecuteOnEnum();
+  const ExecFlagEnum & transfer_execute_on = _pars.get<ExecFlagEnum>("execute_on");
+
+  if (!object_execute_on.contains("TIMESTEP_END") &&
+      !transfer_execute_on.contains("TIMESTEP_BEGIN"))
```

Comment:
this is a tad too loose. The default is 
PP = TIMESTEP_END only and MULTIAPPS=TIMESTEP_BEGIN only

having mroe exec_on on the PP is not a problem
but having more exec_on on the multiapp could mean trouble if they dont match the PP execute on

Reply:
``` if (!object_execute_on.contains("TIMESTEP_END") &&
      !transfer_execute_on.contains("TIMESTEP_BEGIN") && 
      transfer_execute_on.size() <= 1)
?

Reply:
no this is too restrictive now. what if we have 2 execute ons on the transfer

we need to be matching both

Reply:
er sorry I meant ```!transfer_execute_on.size() <= 1)```, so basically, it only triggers if timestep_begin is the only one in the transfer_execute_on

Reply:
I believe that should work, I'll try it out